### PR TITLE
Fix issues with ointment, ailments, and medical stacks

### PR DESCRIPTION
**Pull request by out-of-phaze**
## Description of changes
Makes treating ailments take priority over other interactions, to avoid conflicts with organ treatment.
Replaces `LAZYREMOVE` with `qdel` in `remove_ailment()`.
Removes ailments on organ rejuvenation.
Removes a redundant `use(1)` in ointment code.

## Why and what will this PR improve
Fixes being unable to treat rashes and sprains using ointment and gauze.
Fixes ointment being twice as expensive, using (up to) two stacks per interaction.
Fixes removing ailments via VV not ending ailment effects.
Fixes rejuvenation not removing ailments.

### CODE REVIEW

File: code/_onclick/item_attack.dm [hunk 1]
```diff
@@ -58,6 +58,14 @@ avoid code duplication. This includes items that may sometimes act as a standard
 	if(!ismob(user))
 		return TRUE
 
+	if(!QDELETED(used_item) && user.a_intent == I_HELP)
+		var/obj/item/organ/external/E = GET_EXTERNAL_ORGAN(src, user.get_target_zone())
+		if(length(E?.ailments))
+			for(var/datum/ailment/ailment in E.ailments)
+				if(ailment.treated_by_item(used_item))
+					ailment.was_treated_by_item(used_item, user, src)
+					return
```
Comment (MistakeNot4892) [inline code/_onclick/item_attack.dm]: Should this return TRUE?

File: code/_onclick/item_attack.dm [hunk 2]
```diff
@@ -76,24 +84,10 @@ avoid code duplication. This includes items that may sometimes act as a standard
 
 	var/oldhealth = current_health
 	. = used_item.use_on_mob(src, user)
-	if(used_item.get_attack_force(user) && istype(ai) && current_health < oldhealth)
+	if(used_item.get_attack_force(user) && user != src && istype(ai) && current_health < oldhealth)
```
Comment (MistakeNot4892) [inline code/_onclick/item_attack.dm]: I would suggest doing this in ai.retaliate() itself

File: code/modules/organs/organ.dm [hunk 3]
```diff
@@ -522,12 +523,12 @@ var/global/list/ailment_reference_cache = list()
 	if(ispath(ailment, /datum/ailment))
 		for(var/datum/ailment/ext_ailment in ailments)
 			if(ailment == ext_ailment.type)
-				LAZYREMOVE(ailments, ext_ailment)
+				qdel(ext_ailment)
```
Comment (MistakeNot4892) [inline code/modules/organs/organ.dm:526]: Shouldn't this be in addition to the LAZYREMOVE?

File: code/_onclick/item_attack.dm (hunk 2 above)
Comment (out-of-phaze) [reply code/_onclick/item_attack.dm]: i did not mean to commit that oops

File: code/_onclick/item_attack.dm (hunk 1 above)
Comment (out-of-phaze) [reply code/_onclick/item_attack.dm]: yep

File: code/modules/organs/organ.dm (hunk 3 above)
Comment (out-of-phaze) [reply code/modules/organs/organ.dm:526]: no, `/datum/ailment/Destroy()` removes it from the list. this is exactly how the treatment procs remove them too

**Review by MistakeNot4892 — APPROVED** [review]

**Pull request merged**